### PR TITLE
Add piv build dependencies

### PR DESCRIPTION
**Pull request by Joerger**
[piv-go](https://github.com/ericchiang/piv-go#installation) on Linux depends on [PCSC lite](https://pcsclite.apdu.fr/). This PR introduces this dependency and statically links it in releases.

Prerequisite for https://github.com/gravitational/teleport/pull/15874 and https://github.com/gravitational/teleport/pull/15335

Changes:
  - Add `LIBPCSCLITE_BUILD_TAG` and `LIBPCSCLITE_TEST_TAG`.
  - Add libpcsclite static linking, so that releases do not depend on the dynamic `libpcsclite.so` library.
    - This is only done for Centos7 (`build.assets/Dockerfile-centos7`) because as I understand from https://github.com/gravitational/teleport/pull/14062, we now build our releases with Centos7.
    - `libpcsclite-dev` remains in `build.assets/Dockerfile` for tests, but other pcsc dependencies added in https://github.com/gravitational/teleport/pull/16083/files have been removed in favor of the static version.
    - `LIBPCSCLITE=dynamic` can be used to link to the dynamic `libpcsclite.so` library (local builds)

e PR - https://github.com/gravitational/teleport.e/pull/526

### CODE REVIEW

File: build.assets/Dockerfile-centos7 [hunk 1]
```diff
@@ -92,6 +92,32 @@ RUN mkdir -p /opt && cd /opt && \
...
+RUN yum groupinstall -y 'Development Tools' && \
+    yum install -y \
+        autoconf \
+        libudev-devel && \
+    yum update -y && \
+    yum clean all
+
+# Install libpcsclite.
+# Depends on autoconf.
+# We use outdated version 1.8.11 for licence compatibility.
```
Comment (zmb3) [inline build.assets/Dockerfile-centos7:109]:
```suggestion
# We use outdated version 1.8.11 for license compatibility.
```

**Review by tigrato — APPROVED** [review]
LGTM

File: Makefile [hunk 2]
```diff
@@ -35,63 +35,36 @@ TELEPORT_DEBUG ?= false
...
-endif
 
 OS ?= $(shell go env GOOS)
 ARCH ?= $(shell go env GOARCH)
 FIPS ?=
 RELEASE = teleport-$(GITTAG)-$(OS)-$(ARCH)-bin
 
 # FIPS support must be requested at build time.
-FIPS_MESSAGE := "without FIPS support"
+FIPS_MESSAGE := without-FIPS-support
```
Comment (jakule) [inline Makefile:45]: Why did you add `-` here?

File: build.assets/Dockerfile-centos7 [hunk 3]
```diff
@@ -92,6 +92,31 @@ RUN mkdir -p /opt && cd /opt && \
...
+        autoconf-archive \
+        libudev-devel \
+        scl-utils \
+        centos-release-scl \
+        devtoolset-11-gcc* && \
+    yum clean all
+
+# Install libpcsclite - compile with a newer GCC. The one installed by default is not able to compile it.
+ARG LIBPCSCLITE_VERSION
+RUN git clone --depth=1 https://github.com/gravitational/PCSC.git && \
```
Comment (jakule) [inline build.assets/Dockerfile-centos7:113]: Can you create a tag in our fork and pin the version here? Otherwise, we won't know which version was used for each build.

File: build.assets/Makefile [hunk 4]
```diff
@@ -36,6 +36,7 @@ NODE_VERSION ?= 16.13.2
 RUST_VERSION ?= 1.63.0 # don't bump this without checking GLIBC compatibility
 BORINGCRYPTO_RUNTIME=$(GOLANG_VERSION)b7
 LIBBPF_VERSION ?= 0.7.0-teleport
+LIBPCSCLITE_VERSION ?= 1.9.8
```
Comment (jakule) [inline build.assets/Makefile:39]: Can you rename the tag to `1.9.8-teleport` to indicate that we're not using the mainstream version? The same as we did with `libbpf` line above.

File: Makefile [hunk 5]
```diff
@@ -164,24 +136,44 @@ endif
...
 # from other systems.
-TOUCHID_MESSAGE := without Touch ID
+TOUCHID_MESSAGE := without-Touch-ID
 ifeq ("$(TOUCHID)", "yes")
-TOUCHID_MESSAGE := with Touch ID
+TOUCHID_MESSAGE := with-Touch-ID
 TOUCHID_TAG := touchid
 endif
 
+# Enable libpcsclite for testing?
```
Comment (jakule) [inline Makefile:157]: Why does the "testing" build links dynamically and "non-testing" statically? If we want to avoid this dependency, we could just disable the tag from the build to not force everyone to install it.

File: Makefile [hunk 6]
```diff
@@ -164,24 +136,44 @@ endif
...
+# Eargerly enable if we detect the package, we want to test as much as possible.
+ifeq ("$(shell pkg-config libpcsclite 2>/dev/null; echo $$?)", "0")
+LIBPCSCLITE_TEST_TAG := libpcsclite
+endif
+
+# Build teleport/api against libsclite?
+# Only build if LIBPCSCLITE=yes.
+# This is used for PIV functionality.
+LIBPCSCLITE_MESSAGE := without-PIV-support
+ifneq (, $(filter $(LIBPCSCLITE), yes static dynamic))
```
Comment (jakule) [inline Makefile:167]: Why are you checking the message here? You could check the `LIBPCSCLITE_BUILD_TAG` here and check if it's set already.

**Review by jakule — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (Joerger) [reply Makefile:45]: This commit 7937f9824f55858cbd20c5bd4e2b0962e3c58e0c parses the messages with dashes to produce a more readable build message. This was in response to this comment - https://github.com/gravitational/teleport/pull/15335#discussion_r961241934

File: Makefile (hunk 5 above)
Comment (Joerger) [reply Makefile:157]: The `LIBPCSCLITE_TEST_TAG` build tag is only used by `make test-go` and `make -C build.assets test` to dynamically link to `libpcsclite` for tests. `LIBPCSCLITE_TEST_TAG` does not introduce the dependency when building binaries. This is the same logic used for `libfido2` with the `LIBFIDO2_TEST_TAG` tag.

This isn't necessary but in my opinion it's a nice piece of helper logic to test `libpcsclite` when dynamic linking is possible and static linking is not set up, primarily for local developement. Let me know if I'm missing something though.

**Review by jakule — APPROVED** [review]
